### ISSUE TITLE
Keyboard focus order for Sidebar and Search results should match visual order

### ISSUE THREAD

**Issue by alundgard** [sul-dlss/exhibits#2466]
### Current behavior
The programmatic order (DOM, focus order) does not match the visual order.

In DOM-order, the _Limit your search_ filters ("sidebar" section) comes after _Search results_ ("content" section), but appears before the _Search results_ in visual L-R reading order.

See the following WCAG techniques:
- [G59: Placing the interactive elements in an order that follows sequences and relationships within the content](https://www.w3.org/WAI/WCAG21/Techniques/general/G59)
- [C27: Making the DOM order match the visual order](https://www.w3.org/WAI/WCAG21/Techniques/css/C27)

### Steps to reproduce

WCAG Level A: [Understanding Success Criterion 2.4.3: Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html).

Example page: [Parker Library - Search results - List view](https://exhibits.stanford.edu/parker/catalog?f%5Bformat_main_ssim%5D%5B%5D=Archive%2FManuscript&search_field=manuscript_number&sort=title_sort+asc%2C+pub_year_isi+desc&view=list).

Use keyboard Tab/Shift+Tab to navigate the page. Traverse all _Search results_ before keyboard focus lands on the _Limit your search_ filters.

### Possible solution

Switch the DOM position of `<section id="contents">` and `<section id="sidebar">` so that the sidebar filters precede the search results.

**Note**: The same "sidebar" and "content" arrangement is part of Blacklight and appears in other applications, including [Arclight Beta](https://archives.stanford.edu/collections). The same issue is already corrected in Argo: https://github.com/sul-dlss/argo/issues/4161.

### Reported by SODA
> Violation: Ensure the focus order of interactive elements on the page is logical
> Severity: Major
>
> [Issue]
> The focus order of elements on the page does not match the expected focus order. When at 200%, the filters becomes collapsed into a "hamburger" menu just above the search results. However, this element is after the search results in the focus order, meaning a user goes through all the search results and then jumps back to the top of the results to reach this control. 
>  
> [User Impact]
> When keyboard focus doesn't match the expected focus order, keyboard users will have difficulty interacting with and accessing the controls on the web site.
> 
> [Recommendation]
> Move the menu code to be above the search results in the DOM.

**Comment by taylor-steve** [comment]
From our meeting today this is my understanding of what people thought we should do (@dnoneill 
 & @jcoyne please correct me if I'm wrong here):

* Make this change in the [blacklight layout](https://github.com/projectblacklight/blacklight/blob/dcd30e85af784e91390b62aefa5397406916876d/app/views/layouts/blacklight.html.erb)
* Remove the [spotlight layout](https://github.com/projectblacklight/spotlight/blob/fb0768eb3c5a3a673c1d0a8b78f29d6ec7687071/app/views/layouts/spotlight/spotlight.html.erb) and uses of it

There are differences between the layouts that we'll need to decide how to handle:
* Spotlight has `content?` rather than `content_for?`
* Spotlight has `id="content"`
* Spotlight has `aside` instead of `section` for the sidebar
* Spotlight has `<%= render template: "layouts/spotlight/base" %>` rather than `<%= render template: "layouts/blacklight/base" %>`
